ENIGMAsystem/SHELL/Universal_System/Extensions/DataStructures/data_structures.cpp

```diff
@@ -1240,7 +1240,7 @@ void ds_list_delete(const unsigned int id, const unsigned int first, const unsig
   //Deletes the values in the range between first and last
   if (first < ds_lists[id].size() && last < ds_lists[id].size())
   {
-    ds_lists[id].erase(ds_lists[id].begin() + first, ds_lists[id].begin() + last);
+    ds_lists[id].erase(ds_lists[id].begin() + first, ds_lists[id].begin() + last+1);
```

Comment:
This is correct because vector::erase is exclusive and does not include the last element. We want this ranged delete to be closed since we require the last index to be less than the size.
http://www.cplusplus.com/reference/vector/vector/erase/